x-pack/platform/plugins/shared/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/hooks/navigation.tsx

```diff
@@ -86,38 +86,56 @@ export const useOnSaveNavigate = (params: UseOnSaveNavigateParams) => {
         return;
       }
       const hasNoAgentPolicies = policy.policy_ids.length === 0;
-      const packagePolicyPath =
-        hasNoAgentPolicies || policy.supports_agentless
-          ? getPath('integration_details_policies', {
-              pkgkey: pkgKeyFromPackageInfo(policy.package!),
-            })
-          : getPath('policy_details', {
-              policyId: policy.policy_ids[0], // TODO navigates to first policy
-            });
-      const [onSaveNavigateTo, onSaveQueryParams]: [
-        Parameters<ApplicationStart['navigateToApp']>,
-        CreatePackagePolicyRouteState['onSaveQueryParams']
-      ] = routeState?.onSaveNavigateTo
-        ? [routeState.onSaveNavigateTo, routeState?.onSaveQueryParams]
-        : [
-            [
-              hasNoAgentPolicies || policy.supports_agentless ? INTEGRATIONS_PLUGIN_ID : PLUGIN_ID,
-              {
-                path: packagePolicyPath,
-              },
-            ],
+      let onSaveNavigateTo: Parameters<ApplicationStart['navigateToApp']>;
```

Comment:
most lines here are refactoring for readability (expand inline conditionals as the business logic has gotten more complex)